src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -552,5 +552,15 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor NonConstantPassedToConstantParameter = new DiagnosticDescriptor(
+			id: "D2L0074",
+			title: "Constant parameter cannot be passed a non-constant value.",
+			messageFormat: "Parameter \"{0}\" is marked with the [Constant] attribute, and so cannot be passed a non-constant value.",

```

Comment:
Having part of this be customizable @ the parameter defn would be neat. Like why does it need to be constant? etc.
Not in this PR, could just open an issue for later.

Reply:
https://github.com/Brightspace/D2L.CodeStyle/issues/696